### PR TITLE
Screens to Add and persist repos 

### PR DESCRIPTION
**Pull request by nikhil2611**
## Description
Added the screen to add any chef repository from local system and persist these repositories under My repository options so that they could be available anytime the app opens.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: src/dashboard/index.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                    //   ul.appendChild(li);
+                    // }
+            filesNames.forEach((fileStatus, key) => {
+                console.log(fileStatus, key);
+                var li = document.createElement('li');
+                document.body.appendChild(li);
+                li.innerHTML = fileStatus
+                    ? key + "&nbsp;&nbsp;Active"
+                    : key + "&nbsp;&nbsp;Inactive";
+                var ul = document.getElementById('repos-folder');
```
Comment (vkarve-chef) [inline src/dashboard/index.js:48]: this could be outside the loop

File: src/main.ts [hunk 2]
```diff
@@ -160,18 +160,26 @@ export class Main {
       setTimeout(function () {
         splash.destroy();
       }, 3000);
+    // splash.webContents.openDevTools();
     this.backgroundWindow = new BrowserWindow({
-      show: false,
+      show: true,
```
Comment (vkarve-chef) [inline src/main.ts:165]: is there a way to use feature-flag here? as we merge to main, we don't want this screen to appear until we have the complete flow functional.

File: src/main.ts [hunk 3]
```diff
@@ -222,10 +230,25 @@ export class Main {
...
+ private checkForDuplicate(args){
+    // chef if args path and file name are not already present
+   const path = args
+   // cookbook name can be added later
+   // const cookbook = path.match(/([^\/]*)\/*$/)
+    const obj = helpers.readRepoPath();
+   let status = false
+   for (var i=0; i<obj.length; i++) {
+     if (obj[i]["filepath"] == path) {
+       status = true
```
Comment (vkarve-chef) [inline src/main.ts:242]: can break here

File: src/main.ts [hunk 4]
```diff
@@ -222,10 +230,25 @@ export class Main {
     this.backgroundWindow = null;
     app.quit();
   }
+ private checkForDuplicate(args){
```
Comment (vkarve-chef) [inline src/main.ts:233]: would it make sense to create a `helpers/helpers.ts` and move this method there? or into existing `helpers/helpers.js` if possible?

File: src/main.ts [hunk 5]
```diff
@@ -237,7 +260,43 @@ export class Main {
...
+            }
+          if (!this.checkForDuplicate(filePaths[0])){
+            console.log("returning new cookbook from here")
+            // append this file path to json
+            helpers.writeRepoPath( filePaths[0], "local")
+            // send back cookname and append it to file or refresh whole page
+            event.reply("select-dirs-response", filePaths[0])
+            return filePaths[0]
+          } else{
+            console.log("folder is already present")
```
Comment (vkarve-chef) [inline src/main.ts:292]: can we make the return arg more structured so we can easily discern between success and failure (in this case duplicate folder)?

File: src/process.js [hunk 6]
```diff
@@ -1,2 +1,28 @@
 // Copied from https://github.com/hovancik/stretchly/blob/master/app/process.js - I think we should
 // update our app to have architecture similar to that.
+
+fetch("././dashboard/index.html")
```
Comment (vkarve-chef) [inline src/process.js:4]: `./` needed twice?

File: src/main.ts [hunk 7]
```diff
@@ -237,7 +260,43 @@ export class Main {
...
+            event.reply("select-dirs-response", "folder is already present")
+          }
+              }
```
Comment (vkarve-chef) [inline src/main.ts:295]: fixing indentation all across will improve readability.

**Review by vkarve-chef — CHANGES_REQUESTED** [review]

File: src/dashboard/index.js (hunk 1 above)
Comment (nikhil2611) [reply src/dashboard/index.js]: Done

File: src/helpers/helpers.js [hunk 8]
```diff
@@ -56,10 +58,43 @@ function ExternalAssetsDir() {
   return path.resolve(path.join(ExternalResourcesDir(), "assets"))
 }
 
-module.exports.SrcDir = SrcDir;
+
+function createChefReposJson() {
+  console.log('Creating chef repos.json file')
+  let os = require('os'),
```
[the remark this answers is not in the document]
Comment (nikhil2611) [reply src/helpers/helpers.js]: moved the definition outside of the scope

File: src/process.js (hunk 6 above)
Comment (nikhil2611) [reply src/process.js]: Update to `fetch("./dashboard/index.html")`

File: src/main.ts (hunk 7 above)
Comment (nikhil2611) [reply src/main.ts:295]: fixed.

Comment (nikhil2611) [reply src/main.ts]: Done

File: src/main.ts (hunk 4 above)
Comment (nikhil2611) [reply src/main.ts]: yeah make sense, move this function to `helpers/helpers.js`

**Review by i5pranay93 — APPROVED** [review]

File: src/main.ts (hunk 5 above)
Comment (i5pranay93) [reply src/main.ts]: Yes, returned something like  `event.reply("select-dirs-response", filePaths[0]);` which is received at renderer and is shown at alert.

File: src/main.ts (hunk 3 above)
Comment (i5pranay93) [reply src/main.ts]: Yes, it will stop the extra iteration.

File: src/main.ts (hunk 2 above)
Comment (i5pranay93) [reply src/main.ts:191]: If we do show false, this screen will be hidden.

**Pull request merged**